app/pulp/app/models/progress.py

```diff
@@ -70,14 +78,26 @@ def save(self, *args, **kwargs):
         """
         if self.task_id is None:
             self.task_id = Task.objects.get(id=util.get_current_task_id())
-        super(ProgressReport, self).save(*args, **kwargs)
+
+        now = timezone.now()
+
+
+        if self._using_context_manager and self._last_save_time:
+            if now - self._last_save_time >= datetime.timedelta(milliseconds=BATCH_INTERVAL):
+                super(ProgressReport, self).save(*args, **kwargs)
+        else:
+            super(ProgressReport, self).save(*args, **kwargs)
+
+        self._last_save_time = now
 
     def __enter__(self):
         """
         Saves the progress report state as RUNNING
         """
         self.state = self.RUNNING
         self.save()
+        self._using_context_manager = True
+        return self
```

Comment:
After thinking through this some, I think we do want this. Actually it would allow the docstrings to be improved some to, but you don't have to do that now if you don't want to. Specifically with this change an example that was:

```
        >>> progress_bar = ProgressBar(message='Publishing files', total=len(files_iterator))
        >>> progress_bar.save()
        >>> with progress_bar:
        >>>     for file in progress_bar.iter(files_iterator):
        >>>         handle(file)
```

Could become:

```
        >>> with ProgressBar(message='Publishing files', total=len(files_iterator)) as progress_bar:
        >>>     for file in progress_bar.iter(files_iterator):
        >>>         handle(file)
```